sourced/ml/algorithms/id_splitter/pipeline.py

```diff
@@ -0,0 +1,274 @@
+import argparse
+from datetime import datetime
+import logging
+import pickle
+import random
+import os
+from typing import Callable, Iterable, Tuple
+import warnings
+
+import numpy
+from keras import backend as kbackend
+from keras.callbacks import CSVLogger, TensorBoard, ModelCheckpoint, LearningRateScheduler
+try:
+    import tensorflow as tf
+except ImportError:
+    warnings.warn("Tensorflow is not installed, dependent functionality is unavailable.")
+
+from sourced.ml.algorithms.id_splitter import prepare_features
+
+
+EPSILON = 10 ** -8
```

Comment:
Would be nice to have a comment here which describes where it is needed and how it influences